compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -241,7 +241,7 @@ impl<K: DepKind> DepGraph<K> {
         cx: Ctxt,
         arg: A,
         task: fn(Ctxt, A) -> R,
-        hash_result: impl FnOnce(&mut Ctxt::StableHashingContext, &R) -> Option<Fingerprint>,
+        hash_result: fn(&mut Ctxt::StableHashingContext, &R) -> Option<Fingerprint>,
```

Comment:
This might be another one to back out temporarily to see if it's causing the regression.